protobuf/coreapi/stream.go

```diff
@@ -0,0 +1,38 @@
+package coreapi
+
+import (
+	fmt "fmt"
+
+	grpc "google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	statusKey   = "status"
+	statusReady = "ready"
+)
+
+// WaitForStreamToBeReady waits until the server publish a header with status ready on the stream.
+func WaitForStreamToBeReady(stream grpc.ClientStream) error {
+	// wait for header.
+	header, err := stream.Header()
+	if err != nil {
+		return err
+	}
+	// header received. check status.
+	statuses := header.Get(statusKey)
+	statusesLen := len(statuses)
+	if statusesLen == 0 {
+		return fmt.Errorf("stream header does not contain any status")
+	}
+	lastStatus := statuses[statusesLen-1]
```

Comment:
is there a reason why we have multiple statuses and getting the only last one here?